### PR TITLE
Minor chart additions (ingressClassName + imagePullSecrets)

### PR DESCRIPTION
**Pull request by twiechert**
adding optional value ingressClassName to be used in more recent k8 versions (mostly when using multiple ingress controllers)

### CODE REVIEW

**Comment by VictorKuenstler** [comment]
Hello @twiechert, please package the chart using `helm package .` in the `helm-chart` directory.

**Comment by twiechert** [comment]
Oh yes, thanks for the hint.

I took the freedom to another optional field commonly used (imagePullSecrets).

Also, I wasn't sure about the chart versioning flow applied here, but from my understanding, it would otherwise override the current 0.1.0 (upon helm repo update), thus I bumped the minor version.

**Comment by VictorKuenstler** [comment]
> Oh yes, thanks for the hint.
> 
> I took the freedom to another optional field commonly used (imagePullSecrets).
> 
> Also, I wasn't sure about the chart versioning flow applied here, but from my understanding, it would otherwise override the current 0.1.0 (upon helm repo update), thus I bumped the minor version.

We version the chart by releasing a new tag after merging it into master. But bumping the version additionally is fine.

Somehow, the diff includes the diff of https://github.com/bakdata/streams-explorer/pull/89, could you please check?

**Comment by twiechert** [comment]
hey @VictorKuenstler,

I actually rebased the MR, I'm not sure why the diff view indicates the changes twice - but that should be taken by the squash option in any case, or?

File: helm-chart/templates/ingress.yaml [hunk 1]
```diff
@@ -20,6 +20,9 @@ metadata:
 {{ toYaml . | indent 4 }}
 {{- end }}
 spec:
+{{- if .Values.ingress.ingressClassName }}
+  ingressClassName: {{.Values.ingress.ingressClassName}}
```
Comment (VictorKuenstler) [inline helm-chart/templates/ingress.yaml:24]: Please add an example for the `ingressClassName` in the `values.yaml` as a comment.

**Review by VictorKuenstler — CHANGES_REQUESTED** [review]

File: helm-chart/values.yaml [hunk 2]
```diff
@@ -66,6 +66,9 @@ ingress:
   tls: []
   # Redirect http to https: https://kubernetes-sigs.github.io/aws-load-balancer-controller/v2.1/guide/tasks/ssl_redirect/
   sslRedirect: true
+  # Commonly used to map ingres objects to specific ingress controller installations within your cluster
```
Comment (VictorKuenstler) [inline helm-chart/values.yaml:69]: Minor, but there is a typo for ingress.

**Review by VictorKuenstler — APPROVED** [review]
LGTM, thanks for your contribution @twiechert!

**Comment by twiechert** [comment]
one more side remark:

based on the rebuilt I deleted the old helm chart.

I wonder if it might be more appropriate to retain the old chart artifacts in the master.

reason being:
our vendir based pipeline failed syncing our vendor charts, since a new streams explorer helm chart had been releases and the previous one being referenced was no longer accessible through the master branch.

**Comment by philipp94831** [comment]
> one more side remark:
> 
> based on the rebuilt I deleted the old helm chart.
> 
> I wonder if it might be more appropriate to retain the old chart artifacts in the master.
> 
> reason being:
> our vendir based pipeline failed syncing our vendor charts, since a new streams explorer helm chart had been releases and the previous one being referenced was no longer accessible through the master branch.

You can always reference tags or commits instead of master branch when adding the repo. In the future, we will hopefully release the chart to a proper helm chart repo and this should no longer be an issue

**Comment by twiechert** [comment]
yep this is what we have done now, but I agree a proper chart repo will be the go-to solution anyways.

**Pull request merged**